src/main/java/org/motechproject/mots/web/LocationController.java

```diff
@@ -33,4 +38,56 @@
 
     return locationMapper.toDistrictDtos(districts);
   }
+
+  /**
+   * Get list of districts for preview.
+   * @return list of all districts
+   */
+  @RequestMapping(value = "/districtsOnly", method = RequestMethod.GET)
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public Set<LocationPreviewDto> getDistrictsOnly() {
```

Comment:
did you think about the way to reuse the already fetched list (this used in chw and incharges)?

Reply:
there are 4 separate tabs for each type of location, and the list used in chw and incharges is fully hierarchic